flake.nix

```
@@ -0,0 +1,191 @@
+{
+  inputs = {
+    nixpkgs.url = "nixpkgs/release-23.11";
+    systems.url = "github:nix-systems/default";
+    devenv.url = "github:cachix/devenv";
+    devenv.inputs.nixpkgs.follows = "nixpkgs";
+  };
+
+  nixConfig = {
+    extra-trusted-public-keys = "devenv.cachix.org-1:w1cLUi8dv3hnoSPGAuibQv+f9TZLr6cv/Hm9XgU50cw=";
+    extra-substituters = "https://devenv.cachix.org";
+  };
+
+  outputs = { self, nixpkgs, devenv, systems, ... } @ inputs:
+    let
+      pkg-overlay = (final: prev: let
+        lib = nixpkgs.lib;
+        pkgs = prev;
+      in rec {
+        # Update ndn-cxx to newest commit (required by ndn-svs).
+        ndn-cxx = prev.ndn-cxx.overrideAttrs (old: rec {
+          src = prev.fetchFromGitHub {
+            owner = "named-data";
+            repo = "ndn-cxx";
+            rev = "18ccbb3b1f600d913dd42dd5c462afdac77e37e0";
+            hash = "sha256-yHsp6dBq2kMsubJrn77qeQ9Ah+Udy7nE9eWBX2smemA="; 
+            fetchSubmodules = true; 
+          };
+
+        });
+
+        # Update nfd to newest commit.
+        nfd = prev.nfd.overrideAttrs (old: {
+          src = prev.fetchFromGitHub {
+            owner = "named-data";
+            repo = "nfd";
+            rev = "95d63b113219d1f6bed9fb8f0fff86c9b24db422";
+            hash = "sha256-3TxX9cscbysDRVE5Zr8KHYwlrbI0gkuMldnFLVk9L48=";
+            fetchSubmodules = true;
+          };
+
+          wafConfigureFlags = [
+            "--boost-includes=${prev.boost179.dev}/include"
+            "--boost-libs=${prev.boost179.out}/lib"
+          ];
+          
+          doCheck = false;
+        });
+
+        # Add psync build dependency.
+        psync = lib.makeOverridable prev.stdenv.mkDerivation rec {
+          pname = "psync";
+          version = "8c7c22804c2437166af14156b6681a245e8724fa";
+
+          src = prev.fetchFromGitHub {
+            owner = "named-data";
+            repo = "PSync";
+            rev = "${version}";
+            sha256 = "sha256-IY3hq06l4MYpNw2GKY9g9nLyY/yvuNKl10BYz/CJJyg=";
+          };
+
+          nativeBuildInputs = with prev; [ pkg-config wafHook python3 ];
+          buildInputs = [ ndn-cxx prev.sphinx prev.openssl ];
+
+          wafConfigureFlags = [
+            "--boost-includes=${prev.boost179.dev}/include"
+            "--boost-libs=${prev.boost179.out}/lib"
+            #"--without-tests"
```

Could this line be removed? Or should we keep it?

```suggestion
            #"--without-tests"
```

---

I'd keep it, we should probably re-enable tests as soon as they no longer fail on macOS, and keeping the flag in makes it easier to re-enable.